syn/routes/api/v1/utils.py

```diff
@@ -70,3 +72,34 @@ def tokens():
                 res[chain][token].update({'cgid': cgid})
 
     return res
+
+
+@utils_bp.route('/token_price', methods=['GET'])
+@cache.cached()
+def token_price():
+    chain_id = request.args.get('token', None)
+    token = request.args.get('token', None)
+    date = request.args.get('date', None)
+
+    # validate chain id
+    if not verify.isdigit(chain_id) or (chain_id := int(chain_id) not in CHAINS_REVERSED):
+        return jsonify({'error': 'invalid chain id'}), 400
+    chain_name = CHAINS_REVERSED[chain_id]
+
+    # validate token address
+    if token not in ADDRESS_TO_CGID[chain_name]:
+        return jsonify({'error': 'token for chain not supported'}), 400
+
+    # validate date format if entered
+    if date:
```

Comment:
`if date is None`

Reply:
No ? If date, then parse date and get historical price else get current price

Reply:
my bad, meant `if date is not None`